arviz/plots/lmplot.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+
+from ..rcparams import rcParams
+from .plot_utils import get_plotting_function
+
+
+def plot_lm(
+    y_model,
+    x,
+    y_data=None,
+    y_ppc=None,
+    data=None,
+    x_group=None,
+    y_ppc_group=None,
+    y_data_group=None,
+    num_ppc_samples=100,
+    backend=None,
+):
+    """Plot lm
+    Parameters
+    ----------
+    y_model : str or Sequence
+    x : str or Sequence
+    data : az.InferenceData object
+        Optional if x_data, y_data, y_model and y_ppc are sequence
+    y_data : str or Sequence
+    y_ppc : str or Sequence
+    num_ppc_samples : int
+    x_group : str
+    y_data_group : str
+    y_ppc_group : str
+    backend : str, optional
+        Select plotting backend {"matplotlib","bokeh"}. Default "matplotlib".
+    line_err : bool
+    y_err : bool
+    x_err : bool
+    xscale : str
+    yscale : str
+
+    Returns
+    ------
+    axes
+    """
+
+    groups = data._groups
+    if hasattr(data, "observed_data"):
+        groups = ["observed_data"] + [
+            group for group in groups if group != "observed_data"
+        ]
+
+    y_model_values = y_model
+    if isinstance(y_model, str):
+        if data is None:
+            raise ValueError("data must be present")
+
+        lhs, rhs = y_model.split('~')
+        y_funcs = []
+
+        y_var_name = lhs.strip()
+        while '(' in y_var_name:
+            idx = lhs.find('(')
+            y_funcs = y_funcs + [lhs[:idx]]
+            y_var_name = y_var_name[idx+1:-1]
+
+        if y_data is None:
+            y_data = y_var_name
+
+        if y_ppc is None:
+            y_ppc = y_var_name
+
+        y_model_values=0
```

Comment:
I would consider splitting this into another function and testing it independently